lioncore/metametamodel/builtins.json

```diff
@@ -0,0 +1,360 @@
+{
+  "serializationFormatVersion": "1",
+  "languages": [
+    {
+      "key": "LIonCore-M3",
+      "version": "1"
+    }
+  ],
+  "nodes": [
+    {
+      "id": "LIonCore-builtins",
+      "concept": {
+        "language": "LIonCore-M3",
+        "version": "1",
+        "key": "Language"
+      },
+      "properties": [
+        {
+          "property": {
+            "language": "LIonCore-builtins",
+            "version": "1",
+            "key": "LIonCore-builtins-INamed-name"
+          },
+          "value": "LIonCore.builtins"
+        },
+        {
+          "property": {
+            "language": "LIonCore-M3",
+            "version": "1",
+            "key": "Language-version"
+          },
+          "value": "1"
+        },
+        {
+          "property": {
+            "language": "LIonCore-M3",
+            "version": "1",
+            "key": "IKeyed-key"
+          },
+          "value": "LIonCore-builtins"
+        }
+      ],
+      "children": [
+        {
+          "containment": {
+            "language": "LIonCore-M3",
+            "version": "1",
+            "key": "Language-elements"
+          },
+          "children": [
+            "LIonCore-builtins-String",
+            "LIonCore-builtins-Boolean",
+            "LIonCore-builtins-Integer",
+            "LIonCore-builtins-JSON",
+            "LIonCore-builtins-Node",
+            "LIonCore-builtins-INamed"
+          ]
+        }
+      ],
+      "references": [
+        {
+          "reference": {
+            "language": "LIonCore-M3",
+            "version": "1",
+            "key": "Language-dependsOn"
+          },
+          "targets": []
+        }
+      ],
+      "parent": null
+    },
+    {
+      "id": "LIonCore-builtins-String",
+      "concept": {
+        "language": "LIonCore-M3",
+        "version": "1",
+        "key": "PrimitiveType"
+      },
+      "properties": [
+        {
+          "property": {
+            "language": "LIonCore-builtins",
+            "version": "1",
+            "key": "LIonCore-builtins-INamed-name"
+          },
+          "value": "String"
+        },
+        {
+          "property": {
+            "language": "LIonCore-M3",
+            "version": "1",
+            "key": "IKeyed-key"
+          },
+          "value": "LIonCore-builtins-String"
```

Comment:
For M3 ids are different from keys, while for the builtins they match. For consistency I would make them different, following the same schema used in M3

Reply:
This is something I'm not sure about, see https://lionweb-org.github.io/organization/lioncore/metametamodel/metametamodel.html#_ids_of_built_in_elements

What do you think?

Reply:
I understand that both keys and IDs must be stable, but I do not understand why they need to be identical. However I have just a very tiny preference for them being difficult for consistency with M3 and to catch bugs more easily, but definitely not a strong opinion about this

Reply:
My line of thought: Node ids must only be consistent within one _context_. A context might be one json file, or one repository. Once we move from one context to another (e.g. importing a json file into a repository), all ids might change. That's ok, as long as they change consistently.

(If we later wanted to import another json file that refers to the first one, our repository has to keep the ids of the first one around, but that import issue is not relevant here.)

Keys must be consistent _accross contexts_. Otherwise we always have to ship the M2 along with the M1. (We already decided that.)
This also applies to the keys in `builtins`.

The `builtins` model is special in the sense that we _never_ ship it between contexts -- we always assume it's there.
Thus, we also need stable ids in `builtins`, otherwise we cannot refer to them. But we have to refer to them in every M2.

So yes, we can keep keys and ids different, as long as both are stable.

For `lioncore.json`, the ids (in form `-id-xxx`) don't really matter, as they are gone after import to our repository. We never want to refer to these nodes from the same meta-level (nobody can add/remove things at M3 level). References from other meta-levels use keys.
If we wanted to refer to M3 elements as regular models (e.g. in a documentation language: "Please create a new ->M3.Concept"), these elements are "just" regular nodes, so we refer to them by which id they have in the local context.

For `builtins.json`, the ids _do_ matter. That's why I think the consistency between `lioncore` and `builtins` is not that relevant.

I agree debugging is easier if id differs from key. On the other hand, implementations have to keep two sets of "magic" strings around.

Long story short: I have a slight preference for keeping key == id for `builtins`, but also not a strong opinion.

Reply:
Decision for now: Keep the same. Revisit once we have experience or discover issues.